### DISCUSSION TITLE
Smartypants: Provide a way to configure options?

### DISCUSSION THREAD

**Discussion by JaneSmith** [withastro/roadmap#1027] [Proposal]
### Body

# Summary

Please add a way to configure options for the Smartypants markdown formatter.

# Background & Motivation

Astro uses `remark-smartypants` to format markdown. The plugin can be disabled by setting the `markdown.smartypants` configuration value to `false`. However, it seems there's no way to do any more fine-grained configuration for it. It's just all or nothing. This seems to be inconsistent with another plugin, `remark-reype`, where Astro allows passing options to it via the `markdown.remarkRehype` configuration value.

The [`remark-smartypants` readme](https://github.com/silvenon/remark-smartypants) says it takes the same options as [retext-smartypants](https://github.com/retextjs/retext-smartypants). I would like to be able to configure these options.

Why? Because I may want some of the formatting features, such as converting hyphens to dashes and dots to ellipses, but not others such as the questionable feature of converting apostrophes to single quotes. An individual project may have styling constraints, etc. and so it's useful to be able to control the generated output.

Some examples of people taking issue with the generated output from Smartypants:

- https://github.com/withastro/astro/issues/10859
- https://discourse.gohugo.io/t/smartypants-conversion-issue-with-quotes/47942
- https://linkinpark213.com/2018/03/20/smartypants/
- https://github.com/Python-Markdown/markdown/issues/654

Hugo, an alternative SSG, does seem to allow this more fine-grained control over formatting.

# Goals

- Allow Smartypants options to be defined in the Astro configuration file.

# Example

```
{
  markdown: {
    smartypants: { quotes: false },
  },
};
```

**Comment by delucis** [comment]
Thanks for the feedback! If you're looking for a quick workaround, here's how Astro's docs configure options for smartypants:
https://github.com/withastro/docs/blob/b1772b53c0fa7e912fb50185e40f203cc6ffe99e/astro.config.ts#L76-L79

**Comment by bluwy** [comment]
Seems like a nice addition to me to allow passing an object config.

**Comment by Daiz** [comment]
I would also like to be able to configure smartypants in Astro. The fact that Astro itself is hacking around the lack of such configuration seems to be a good indicator that it should be possible, honestly.

**Reply by florian-lefebvre** [reply]
PR is welcome!

**Reply by trueberryless** [reply]
I created https://github.com/withastro/astro/pull/15340 after I finally got some time to work on this enhancement.
